packages/agoric-cli/lib/start.js

```diff
@@ -494,6 +501,10 @@ export default async function startMain(progname, rawArgs, powers, opts) {
           break;
         }
       }
+      if (!bestRpcAddr) {
+        // eslint-disable-next-line no-await-in-loop
+        await delay(2000);
+      }
```

Comment:
Is there truly no way to listen(0) to get a port assignment and delegate it downward, or listen(0) in a client and communicate it upward over IPC?

Reply:
Not sure how what you're referring to relates to these lines.

This loop is waiting for the local chain to start listening, which may be at different hosts (`host.docker.internal` or `127.0.0.1`) depending on what Docker setup you have, but is always at a specified port.

Reply:
Ah, thanks for clarifying.

The wrinkle is that arbitrary 2s. There’s got to be an operating system analog for a promise that we can wield for these cases, like a read on a named pipe that gets passed as an argument to the daemon and closed by the daemon after it opens its listen socket and is ready to respond. But, such is the nature of these problems that they’re sufficiently hard to nail, and very easy to work-around, that there aren’t established norms. 🤷‍♂️